compiler/backend/src/org/jetbrains/kotlin/codegen/inline/defaultMethodUtil.kt

```diff
@@ -202,3 +179,68 @@ private fun defaultLambdaFakeCallStub(args: Array<Type>, lambdaOffset: Int): Met
         false
     )
 }
+
+fun loadDefaultLambdaBody(classBytes: ByteArray, classType: Type, isPropertyReference: Boolean): SMAPAndMethodNode {
+    // In general we can't know what the correct unboxed `invoke` is, and what Kotlin types its arguments have,
+    // as the type of this object may be any subtype of the parameter's type. All we know is that Function<N>
+    // has to have a `invoke` that takes `Object`s and returns an `Object`; everything else needs to be figured
+    // out from its contents. TODO: for > 22 arguments, the only argument is an array. `MethodInliner` can't do that.
+    val invokeName = if (isPropertyReference) OperatorNameConventions.GET.asString() else OperatorNameConventions.INVOKE.asString()
+    val invokeNode = getMethodNode(classBytes, classType) {
+        it.name == invokeName && it.returnType == AsmTypes.OBJECT_TYPE && it.argumentTypes.all { arg -> arg == AsmTypes.OBJECT_TYPE }
+    } ?: error("can't find erased invoke '$invokeName(Object...): Object' in default lambda '${classType.internalName}'")
+    return if (invokeNode.node.access.and(Opcodes.ACC_BRIDGE) == 0)
+        invokeNode
+    else
+        invokeNode.node.inlineBridge(classBytes, classType)
+}
+
+private fun MethodNode.inlineBridge(classBytes: ByteArray, classType: Type): SMAPAndMethodNode {
+    // If the erased invoke is a bridge, we need to locate the unboxed invoke and inline it. As mentioned above,
+    // we don't know what the Kotlin types of its arguments/returned value are, so we can't generate our own
+    // boxing/unboxing code; luckily, the bridge already has that.
+    val invokeInsn = instructions.singleOrNull { it is MethodInsnNode && it.owner == classType.internalName } as MethodInsnNode?
```

Comment:
Could RemapVisitor be used here to inline instuctions or maybe Inliner itself?

Reply:
I'd prefer not to do that since MethodInliner has a lot of other stuff, and here we pretty much just need to copy-paste bytecode. Even for inlining lambdas in the IR backend MethodInliner is already major overkill, since the only relevant part in it is capture remapping.

Reply:
Indeed there is straitforward 1:1 mapping